swift/ql/lib/codeql/swift/regex/Regex.qll

```diff
@@ -307,15 +312,15 @@ abstract class RegexEval extends CallExpr {
  * A call to a function that always evaluates a regular expression.
  */
 private class AlwaysRegexEval extends RegexEval {
-  Expr regexInput;
-  Expr stringInput;
+  DataFlow::Node regexInput;
+  DataFlow::Node stringInput;
```

Comment:
Very nit: I would have probably kept these as `Expr`s to avoid all those `asExpr()` calls in the charpred, and then simply implemented `getRegexInput` as `result.asExpr() = regexInput`.

(and similarly for the other members of this class and related classes.)

Reply:
I'm _trying_ to move all the interfaces in this library towards `DataFlow::Node` rather than a split between `DataFlow::Node` and `Expr`.  I think this is more natural and robust.  When we get there a lot of these conversions should disappear (though there will no doubt be some `deprecated` stuff for a time).

Reply:
I've now gone ahead with the changes to a `DataFlow::Node`-based interface and deprecated the old one.  Better now rather than later!